modules/scripts/timeline.py

```diff
@@ -0,0 +1,50 @@
+from docutils import nodes
+
+from docutils.parsers import rst
+
+class timeline(nodes.General, nodes.Element):
+  pass
+
+def visit(self, node):
```

Comment:
Then why is the first parameter self? `self` is used as the first param for methods inside a class to access the object. It is similar to `this` in C++